### ISSUE TITLE
Version 2.2.0 MacOS Symbol not found: _iconv

### ISSUE THREAD

**Issue by wdeanHPA** [pymssql/pymssql#705]
Upon upgrade to 2.2.0 from 2.1.x, import of package did not work. 

>>> import pymssql
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/williamdean/Desktop/.../.venv/lib/python3.8/site-packages/pymssql/__init__.py", line 3, in <module>
    from ._pymssql import *
  File "src/pymssql/_pymssql.pyx", line 1, in init pymssql._pymssql
ImportError: dlopen(/Users/williamdean/Desktop/.../.venv/lib/python3.8/site-packages/pymssql/_mssql.cpython-38-darwin.so, 2): Symbol not found: _iconv
  Referenced from: /Users/williamdean/Desktop/.../.venv/lib/python3.8/site-packages/pymssql/_mssql.cpython-38-darwin.so
  Expected in: flat namespace
 in /Users/williamdean/Desktop/.../.venv/lib/python3.8/site-packages/pymssql/_mssql.cpython-38-darwin.so

Using MacOS Catalina Version 10.15.7 tried with python 3.7-3.9. 
>>> tsql -C
Compile-time settings (established with the "configure" script)
                            Version: freetds v1.2.18
             freetds.conf directory: /usr/local/etc
     MS db-lib source compatibility: no
        Sybase binary compatibility: yes
                      Thread safety: yes
                      iconv library: yes
                        TDS version: 7.3
                              iODBC: no
                           unixodbc: yes
              SSPI "trusted" logins: no
                           Kerberos: yes
                            OpenSSL: yes
                             GnuTLS: no
                               MARS: yes

Switched back to pymssql to 2.1.x and everything worked fine.

**Comment by termim** [comment]
How did you upgrade to 2.2.0? In 2.2.0 wheels on PyPi there is no pymssql/init.py file.

**Comment by wdeanHPA** [comment]
> How did you upgrade to 2.2.0? In 2.2.0 wheels on PyPi there is no pymssql/init.py file.

I used `pipenv install` (pipenv, version 2020.11.15) to install pymssql. It took the latest version from `https://pypi.org/simple`

**Comment by termim** [comment]
Could you provide a listing of `/Users/williamdean/Desktop/.../.venv/lib/python3.8/site-packages/pymssql/` directory to verify that `init.py` is there?

**Comment by wdeanHPA** [comment]
ls -la .venv/lib/python3.9/site-packages/pymssql

total 2824
drwxr-xr-x    6 williamdean  staff      192 Apr 13 12:35 .
drwxr-xr-x  134 williamdean  staff     4288 Apr 13 12:39 ..
-rw-r--r--    1 williamdean  staff      101 Apr 13 12:35 __init__.py
drwxr-xr-x    3 williamdean  staff       96 Apr 13 12:35 __pycache__
-rwxr-xr-x    1 williamdean  staff  1072672 Apr 13 12:35 _mssql.cpython-39-darwin.so
-rwxr-xr-x    1 williamdean  staff   366800 Apr 13 12:35 _pymssql.cpython-39-darwin.so

**Comment by termim** [comment]
Where does init.py come from?

**Comment by wdeanHPA** [comment]
it is  _ _ init _ _ .py but github was reformating it. double underscore file

**Comment by fazzzzzzz** [comment]
I also encountered the same problem.
```
$ /usr/local/opt/python@3.9/bin/python3.9 xxx.py

Traceback (most recent call last):
  File "xxx.py", line 1, in <module>
    import pymssql
  File "/usr/local/lib/python3.9/site-packages/pymssql/__init__.py", line 3, in <module>
    from ._pymssql import *
  File "src/pymssql/_pymssql.pyx", line 1, in init pymssql._pymssql
ImportError: dlopen(/usr/local/lib/python3.9/site-packages/pymssql/_mssql.cpython-39-darwin.so, 2): Symbol not found: _iconv
  Referenced from: /usr/local/lib/python3.9/site-packages/pymssql/_mssql.cpython-39-darwin.so
  Expected in: flat namespace
 in /usr/local/lib/python3.9/site-packages/pymssql/_mssql.cpython-39-darwin.so

$ ls /usr/local/lib/python3.9/site-packages/pymssql 

__init__.py                   __pycache__                   _mssql.cpython-39-darwin.so   _pymssql.cpython-39-darwin.so
```

**Comment by termim** [comment]
Do you have libiconv installed?

**Comment by fazzzzzzz** [comment]
```shell
$ iconv --version

iconv (GNU libiconv 1.16)
Copyright (C) 2000-2019 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Written by Bruno Haible.
```

**Comment by pdxJaxon** [comment]
I too am having the same issue....Im dead in the water right now on Pyton development --> Sql Server

**Comment by pdxJaxon** [comment]
File "main.py", line 1, in <module>
    import pymssql
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pymssql/__init__.py", line 3, in <module>
    from ._pymssql import *
  File "src/pymssql/_pymssql.pyx", line 1, in init pymssql._pymssql
ImportError: dlopen(/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pymssql/_mssql.cpython-37m-darwin.so, 2): Symbol not found: _iconv
  Referenced from: /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pymssql/_mssql.cpython-37m-darwin.so
  Expected in: flat namespace
 in /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pymssql/_mssql.cpython-37m-darwin.so
jaxon@Gregorys-MBP pyodbc $

**Comment by opetrenko-grid** [comment]
installing libiconv do not help

**Comment by wdeanHPA** [comment]
> installing libiconv do not help. Downgrading to 2.1.5 fixed the issue

Agreed. It didn't help with me either

**Comment by chuenshyu** [comment]
how do you upgrade or downgrad libiconv to 2.1.5?
is there any easy comment i can do the same?

**Comment by alfred42** [comment]
Same here.

**Comment by alfred42** [comment]
Downgraded to 2.1.5 and it works fine.

**Comment by termim** [comment]
Could you try the wheel from [test.pypi](https://test.pypi.org/project/pymssql/2.2.2.dev7/#files)?

**Comment by chuenshyu** [comment]
macbookpro-2:Python_Scraping chuenshyu$ pip install -i https://test.pypi.org/simple/ pymssql==2.2.2.dev7Copy PIP instructions
Stable version available (2.2.0)
Looking in indexes: https://test.pypi.org/simple/
ERROR: Could not find a version that satisfies the requirement pymssql==2.2.2.dev7Copy (from versions: 2.1.1, 2.1.2.dev1, 2.1.2rc1.dev0, 2.1.2rc1.dev8, 2.1.3.dev1, 2.1.3.dev2, 2.1.4.dev4, 2.1.4rc2, 2.1.4rc3, 2.1.4, 2.1.5, 2.1.6.dev11, 2.1.6.dev12, 2.1.6.dev16, 2.1.6.dev19, 2.1.6.dev20, 2.1.6.dev38, 2.1.6.dev39, 2.1.6.dev41, 2.1.6.dev42, 2.1.6.dev43, 2.1.6.dev55, 2.1.6.dev71, 2.1.6.dev73, 2.1.6.dev75, 2.1.6.dev77, 2.2.0, 2.2.1.dev1, 2.2.1.dev2, 2.2.2.dev3, 2.2.2.dev5, 2.2.2.dev7)
ERROR: No matching distribution found for pymssql==2.2.2.dev7Copy
bash: syntax error near unexpected token `(‘

**Comment by chuenshyu** [comment]
Nevermind, fault positive..


pip install -i https://test.pypi.org/simple/ pymssql==2.2.2.dev7

Run with PYINSTALLER and my app now builds fine without error.

**Comment by termim** [comment]
Should be fixed in edc76e6b.

**Comment by Shlomo-T** [comment]
Hi,
This still happens for me with pymssql 2.2.2. 
Running Python 3.8.9 on macOS.
It seems that a workaround (somehow) is to import _scproxy. I guess that the module contains a symbol for iconv that solves it but that's a very ugly patch.

**Comment by jasonred8** [comment]
@Shlomo-T 

I have same issue , I also using pymssql 2.2.2
when I uninstall pymssql and reinstall pymssql 2.1.5 , It has been resolve ...

```
pip3 uninstall pymssql

pip3 install pymssql==2.1.5 
```

**Comment by KarthikRajashekaran** [comment]
I tried with `pipenv install pymssql` and it failed 
Then tried with `pip install pymssql` but  `import pymssql` not working  in Mac OS (Big Sur 11.5.2)
Whats the solution?

**Comment by porthunt** [comment]
@KarthikRajashekaran Check this: https://stackoverflow.com/questions/44958536/how-to-install-pymssql-on-macos-sierra#answer-66176998

**Comment by EmmanT** [comment]
> Hi,
> This still happens for me with pymssql 2.2.2.
> Running Python 3.8.9 on macOS.
> It seems that a workaround (somehow) is to import _scproxy. I guess that the module contains a symbol for iconv that solves it but that's a very ugly patch.

Confirmed!
import _scproxy fixed it and its ugly indeed.

**Comment by gagarcr** [comment]
> > Hi,
> > This still happens for me with pymssql 2.2.2.
> > Running Python 3.8.9 on macOS.
> > It seems that a workaround (somehow) is to import _scproxy. I guess that the module contains a symbol for iconv that solves it but that's a very ugly patch.
> 
> Confirmed! import _scproxy fixed it and its ugly indeed. im using pymssql 2.2.2 for this in Mac OS Big Sur 11.5.2

Confirmed here too. Just import _scproxy before and it fixes it. Mac Os Big Sur 11.6

**Comment by rliessum** [comment]
> > Hi,
> > This still happens for me with pymssql 2.2.2.
> > Running Python 3.8.9 on macOS.
> > It seems that a workaround (somehow) is to import _scproxy. I guess that the module contains a symbol for iconv that solves it but that's a very ugly patch.
> 
> Confirmed! import _scproxy fixed it and its ugly indeed. im using pymssql 2.2.2 for this in Mac OS Big Sur 11.5.2

Also confirmed using the same solution! (Mac OS Monterey 12.1 & python 3.9.9.

**Comment by HenryGaoGH** [comment]
> Hi,
> This still happens for me with pymssql 2.2.2.
> Running Python 3.8.9 on macOS.
> It seems that a workaround (somehow) is to import _scproxy. I guess that the module contains a symbol for iconv that solves it but that's a very ugly patch.

pymssql Version 2.2.4
double confirm!
It's still ugly

**Comment by anthonygrant** [comment]
> Hi, This still happens for me with pymssql 2.2.2. Running Python 3.8.9 on macOS. It seems that a workaround (somehow) is to import _scproxy. I guess that the module contains a symbol for iconv that solves it but that's a very ugly patch.

Can confirm also. Still an issue in pymssql 2.2.4
Reverting to 2.1.5 until this actually gets resolved. 

Not sure why this is closed when it is still an issue and has never been resolved.

**Comment by VBoB13** [comment]
Because it's MSSQL... XD
Jokes aside, I am also wondering why they never fixed the issue and closed it... really makes you wonder how serious they are as it doesn't seem like the biggest issue...
Maybe they don't want to solve problems for MacOS because it's MS? :P

**Comment by termim** [comment]
> Because it's MSSQL... XD Jokes aside, I am also wondering why they never fixed the issue...

May be because you did not submit PR yet? 
May be because they do not have MacOS?

Chose whatever suits you most ;)

**Comment by VBoB13** [comment]
> > Because it's MSSQL... XD Jokes aside, I am also wondering why they never fixed the issue...
> 
> May be because you did not submit PR yet? May be because they do not have MacOS?
> 
> Chose whatever suits you most ;)

Well, these days, when developing frameworks/libraries like this, it's pretty easy for an expert like yourself to throw the project up on a virtual MacOS server to test things in that environment.
Wouldn't you agree? :)

**Comment by spointeau** [comment]
I confirm that doing 
```
import _scproxy
import pymssql
```
works.

I spent hours finding this, could it be fixed ?

**Comment by Jonghyun-Chung** [comment]
> I confirm that doing
> 
> ```
> import _scproxy
> import pymssql
> ```
> 
> works.
> 
> I spent hours finding this, could it be fixed ?

This worked for me. Thank you.

**Comment by apocalypse2012** [comment]
Please reopen this issue. The tacky stupid work around has me blocked on Docker. I cannot import _scproxy on my docker images.

**Comment by caidingyu** [comment]
This issue still exists in pymssql 2.2.5. (MacOS 12.4, Python 3.9.13).
Reinstall to pymssql 2.1.5 fixed my issue.

Please re-open this.

**Comment by bhshar** [comment]
still this issue exists.

**Comment by yihuifeng1999** [comment]
I am still facing this issue in pymssql 2.2.5, Mac OS 12.6 with python 3.9.7, got worked around by reverting back to 2.1.5 using this:

> @KarthikRajashekaran Check this: https://stackoverflow.com/questions/44958536/how-to-install-pymssql-on-macos-sierra#answer-66176998

Or `import _scproxy` also works.

> Hi, This still happens for me with pymssql 2.2.2. Running Python 3.8.9 on macOS. It seems that a workaround (somehow) is to import _scproxy. I guess that the module contains a symbol for iconv that solves it but that's a very ugly patch.

The funny thing I met is, importing tkinter solves this issue as well. I guess this just somehow does the same thing.

**Comment by softgandhi** [comment]
I am also facing this issue.
is this fixed or should I downgrade to version 2.1.5?

can anyone pls help.

**Comment by kangjian99** [comment]
I reverted it to 2.2.2, solved. Really strange.
2.1.5 caused other problems for me.
